addons/repair/models/repair.py

```diff
@@ -451,7 +451,9 @@ def action_repair_end(self):
         """
         if self.filtered(lambda repair: repair.state != 'under_repair'):
             raise UserError(_("Repair must be under repair in order to end reparation."))
-        if any(float_compare(move.quantity, move.product_uom_qty, precision_rounding=move.product_uom.rounding) < 0 for move in self.move_ids):
+        partial_moves = self.move_ids.filtered(lambda m: float_compare(m.quantity, m.product_uom_qty, precision_rounding=m.product_uom.rounding) < 0)
+        picked_moves = self.move_ids.filtered(lambda m: m.picked)
```

Comment:
```suggestion
        partial_moves = set()
        picked_moves = set()
        for move in self:
             if float_compare(move.quantity, move.product_uom_qty, precision_rounding=m.product_uom.rounding) < 0):
                 partial_moves.add(move.id)
             if move.picked:
                 picked_moves.add(move.id)
```
to avoid 2 loops